File: prep/implementation/patterns_reference_imeplementation/sliding_window/MaximumSumSubarrayOfSizeKBrute.java
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;

// O(n * k)
public class MaximumSumSubarrayOfSizeKBrute {
  public static void main(String[] args) {
    try (Scanner in = new Scanner(System.in)) {
      int tt = in.nextInt();

      while (tt-- > 0) {
        int n = in.nextInt();
        int k = in.nextInt();

        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
          a[i] = in.nextInt();
        }

        int maxSum = 0;
        for (int i = 0; i < n - k + 1; i++) {
          int currSum = 0;

          for (int j = i; j < i + k; j++) {
            currSum += a[j];
          }
          maxSum = Math.max(maxSum, currSum);
        }

        System.out.println(maxSum);
      }
    }
  }
}

File: prep/implementation/patterns_reference_imeplementation/sliding_window/maximum_sum_subarray_of_size_k_brute.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  int tt, n, k;
  cin >> tt;

  while (tt--) {
    cin >> n >> k;

    vector<int> a(n);
    for (int i = 0; i < n; i++) {
      cin >> a[i];
    }

    int max_sum = 0;
    for (int i = 0; i < n - k + 1; i++) {
      int curr_sum = 0;
      for (int j = i; j < i + k; j++) {
        curr_sum += a[j];
      }
      max_sum = max(max_sum, curr_sum);
    }

    cout << max_sum << "\n";
  }

  return 0;
}